frontend/src/components/GrantAccess.vue

```diff
@@ -4,26 +4,44 @@ import { ApiService, type ApplicationRoleResponse, type GrantUserRoleRequest } f
 import { selectedApplication } from '@/services/ApplicationState';
 import { onMounted, ref } from 'vue';
 import { useToast } from 'vue-toastification';
+import { Form as VeeForm, Field, ErrorMessage } from 'vee-validate';
+import router from '@/router';
+// import * as yup from 'yup';
+import { object, string } from 'yup';
 
 const FOREST_CLIENT_INPUT_MAX_LENGTH = 8
-const domainOptions = {IDIR: 'I', BCEID: 'B'} // TODO, load it from backend when backend has the endpoint.
+const domainOptions = { IDIR: 'I', BCEID: 'B' } // TODO, load it from backend when backend has the endpoint.
 let applicationRoleOptions = ref<ApplicationRoleResponse[]>([])
-
 const defaultFormData = {
-  domain: domainOptions.BCEID,
-  userId: null,
-  forestClientNumber: null,
-  role: null as unknown as ApplicationRoleResponse
+    domain: domainOptions.BCEID,
+    userId: null,
+    forestClientNumber: null,
+    role: null as unknown as ApplicationRoleResponse
 }
-const formData = ref(JSON.parse(JSON.stringify(defaultFormData))) // clone default input data.
-
-const apiService = new ApiService()
+const formData = ref({...defaultFormData}); // clone default input data.
```

Comment:
Hi Nick, thanks for change this using destructuring assignment syntax. I was not so familiar with it.
However, I am just wondering, does it do deep-cloning for the object (The 'role' is another level of nested object)?

Reply:
I have switched back to the previous implementation.